conductor_api/src/conductor/base.rs

```diff
@@ -472,6 +511,29 @@ impl Conductor {
         Dna::try_from(JsonString::from(contents))
     }
 
+    /// Default KeyLoader that actually reads files from the filesystem
+    fn load_key(file: &PathBuf) -> Result<Keypair, HolochainError> {
+        notify(format!("Reading agent key from {}", file.display()));
+
+        let mut file = File::open(file)?;
+        let mut contents = String::new();
+        file.read_to_string(&mut contents)?;
+
+        let passphrase = rpassword::read_password_from_tty(Some("Passphrase: "))?;
+
+        let bundle: KeyBundle = serde_json::from_str(&contents)?;
+        let mut passphrase = SecBuf::with_insecure_from_string(passphrase);
```

Comment:
passphrases should use secure memory

Ideally, we'd use a low-level keyboard binding to prevent a class of malware keyloggers and input the data directly into a secure buffer.

For the short-term, we should be sure to zero the memory in "passphrase" returned by the rpassword utility.

Reply:
I had tried that already in `hc keygen` but `SodiumBuf` doesn't allow arbitrary sizes for secure buffers. Only [multiples of 8](https://github.com/holochain/holochain-rust/blob/develop/sodium/src/secbuf.rs#L70). Also, since the passphrase is coming in as a `String`, it wouldn't make it much safer, right?

But yeah, so you are saying, we should roll our own `read_password_from_tty` eventually?

Reply:
: ) right, that totally makes sense. I'll create an issue to do the alignment in the background and deal with appropriately sized slice references in the SecBuf so we don't have to worry about the sizing during implementation.

Reply:
https://github.com/holochain/holochain-rust/issues/1017